src/codec/encoder/MayaEncoder.cpp

```diff
@@ -430,7 +441,27 @@ SerializedGeometry serializeGeometry(const prtx::GeometryPtrVector& geometries,
 		}
 		++matsIt;
 	}
-	SerializedGeometry sg(numCounts, numIndices, maxNumUVSets);
+
+	std::vector<uint32_t> numUvs(maxNumUVSets);
+	std::vector<uint32_t> numUvCounts(maxNumUVSets);
+	std::vector<uint32_t> numUvIndices(maxNumUVSets);
+
+	for (const auto& geo : geometries) {
```

Comment:
we should use this opportunity to hide both this new and the old "counting loop" into the `SerializedGeometry` constructor to hide lots of code which does not contribute to the functionality... probably needs a getter for `maxNumUVSets`...

Reply:
Moved counting loop into constructor of Serialized Geometry.
A getter is not needed, as uvs.size() already corresponds to maxNumUVSets